ui/src/components/head/__inline__/loadFonts.js

```diff
@@ -0,0 +1,376 @@
+// @flow
+
+/*
+bypass normal browser font-loading to avoid the FOIT. works like this:
+
+do you have fonts in localStorage?
+    yes – inject them (minimises 2nd layout as fonts are loaded before DOM is created)
+    no  – did the localStorage check go ok?
+        yes – ajax them in as JSON immediately, inject them and save them to localStorage
+        no  – load font files async using @@font-face
```

Comment:
> no  – load font files async using @@font-face

It’s in this fallback scenario that we could try to leverage [FontFaceSet](https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet), to avoid bad browser behaviour. But then, how likely are they to support it while not supporting localstorage? (Safari in Private Browsing mode?).